dashboard/app/models/pd/application/facilitator1819_application.rb

```diff
@@ -462,5 +467,66 @@ def sanitize_form_data_hash
     def check_idempotency
       Pd::Application::Facilitator1819Application.find_by(user: user)
     end
+
+    before_save :destroy_fit_autoenrollment, if: -> {status_changed? && status != "accepted"}
+    def destroy_fit_autoenrollment
+      return unless auto_assigned_fit_enrollment_id
+
+      Pd::Enrollment.find_by(id: auto_assigned_fit_enrollment_id).try(:destroy)
+      self.auto_assigned_fit_enrollment_id = nil
+    end
+
+    def fit_workshop
+      Pd::Workshop.find(fit_workshop_id) if fit_workshop_id
+    end
+
+    # override
+    def enroll_user
+      super
+      return unless fit_workshop_id
```

Comment:
Is this the same as `WorkshopAutoenrolledApplication.enroll_user`, but with `fit_workshop_id` instead of `pd_workshop_id`? Would it make sense to combine the bulk of this method into a common one, something like `enroll_in_workshop(pd_workshop_id)` and then just override a smaller method that retrieves that id?

Reply:
It is the same, but also with `destroy_fit_autoenrollment` and `auto_assigned_fit_enrollment_id` instead of the pd workshop versions of the same. It was enough difference that making a common method was a pain